homeassistant/components/keba/binary_sensor.py

```diff
@@ -0,0 +1,110 @@
+
+
+async def async_setup_platform(hass, config,
+                               async_add_entities, discovery_info=None):
+    """Set up the KEBA charging station platform."""
+    _LOGGER.debug("Initializing KEBA charging station binary sensors")
+    keba = hass.data[DOMAIN]
+
+    sensors = [
+        KebaBinarySensor(keba, 'Online', 'Wallbox', 'connectifity'),
+        KebaBinarySensor(keba, 'Plug', 'Plug', 'plug'),
+        KebaBinarySensor(keba, 'State', 'Charging state', 'power'),
+        KebaBinarySensor(keba, 'Tmo FS', 'Failsafe Mode', 'safety')
+    ]
+    async_add_entities(sensors)
+
+
+class KebaBinarySensor(BinarySensorDevice):
+    """Representation of a binary sensor of a KEBA charging station."""
+
+    def __init__(self, keba, key, sensor_name, device_class):
+        """Initialize the KEBA Sensor."""
+        self._key = key
+        self._keba = keba
+        self._name = sensor_name
+        self._device_class = device_class
+        self._is_on = None
+        self._attributes = {}
+
+    @property
+    def should_poll(self):
+        """Deactivate polling. Data updated by KebaHandler."""
+        return False
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of the binary sensor."""
+        return self._keba.device_name + '_' + self._name
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def device_class(self):
+        """Return the class of this sensor."""
+        return self._device_class
+
+    @property
+    def is_on(self):
+        """Return true if sensor is on."""
+        return self._is_on
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the binary sensor."""
+        return self._attributes.items()
+
+    async def async_update(self):
+        """Get latest cached states from the device."""
+        if self._key == 'Online':
+            self._is_on = self._keba.get_value(self._key)
+            self._attributes['Product'] = self._keba.get_value('Product')
+            self._attributes['Serial'] = self._keba.get_value('Serial')
+            self._attributes['Firmware'] = self._keba.get_value('Firmware')
+            self._attributes['Device uptime'] = \
+                self._keba.get_value('uptime_pretty')
+
+        elif self._key == 'Plug':
+            self._is_on = self._keba.get_value('Plug_plugged')
+            self._attributes["Plugged on wallbox"] = \
+                self._keba.get_value('Plug_wallbox')
+            self._attributes["Plug locked"] = \
+                self._keba.get_value('Plug_locked')
+            self._attributes["Plugged on EV"] = self._keba.get_value('Plug_EV')
+
+        elif self._key == 'State':
+            self._is_on = self._keba.get_value('State_on')
+            self._attributes['status'] = self._keba.get_value('State_details')
+            self._attributes['max charging rate (A)'] = \
```

Comment:
The state attributes should be template friendly, not look good in the frontend.

Please remove the unit from the key name.

Reply:
fixed